kaldb/src/main/java/com/slack/kaldb/chunk/SearchContext.java

```diff
@@ -0,0 +1,26 @@
+package com.slack.kaldb.chunk;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.slack.kaldb.proto.config.KaldbConfigs;
+
+public class SearchContext {
+  public static SearchContext fromConfig(KaldbConfigs.ServerConfig serverConfig) {
+    return new SearchContext(serverConfig.getServerAddress(), serverConfig.getServerPort());
+  }
+
+  public final String hostname;
+  public final int port;
+
+  public SearchContext(String hostname, int port) {
+    checkState(hostname != null && !hostname.isEmpty(), "hostname field can't be null or empty");
```

Comment:
These should be `checkArgument` instead of `checkState`

Reply:
Damn autocomplete. Done.